framework_tvb/tvb/core/entities/file/simulator/operation_group_h5.py

```diff
@@ -0,0 +1,84 @@
+
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+import uuid
+from tvb.basic.neotraits.api import Attr
+from tvb.core.entities.file.exceptions import MissingDataSetException
+from tvb.core.neotraits.h5 import Json, H5File, Scalar, Reference
+
+
+class OperationGroupH5(H5File):
```

Comment:
This is a bit controversial.
Have you considered instead that on each ViewModel part of a group to simply store the ranges? This was the previous approach (approx) and the redundancy is not much but it preserves the structure without the need to have an odd Group separate serialization

Reply:
Yes, we are still considering simplifying the solution by using only the ViewModel, but I think only ranges are not enough for us to determine which operations are part of the same group, we need some extra information (maybe the GID of the OpGroup).